lib/pbench/server/auth/auth.py

```diff
@@ -130,51 +182,97 @@ def verify_auth(auth_token):
         return None
 
     @staticmethod
-    def verify_third_party_token(auth_token: str, oidc_client: OpenIDClient) -> bool:
+    def verify_third_party_token(
+        auth_token: str,
+        oidc_client: OpenIDClient,
+        algorithms: Optional[list[str]] = ["HS256"],
+    ) -> "InternalUser":
```

Comment:
> * It is bad practice to use non-scalars as default parameter values.  The problem is that the value is reused on each call to the function; thus, if the (here) list is mutated during the function execution, then the _next_ call to the function which uses the default will receive the **mutated** value.  (This makes sense if you think deeply about it, because the value is the _reference_ to the list and not the contents of the list, but it is superficially counterintuitive and therefore often produces surprising results.)

That is interesting, I didn't think about that. Thank you, I will do some experimentation with this.

Reply:
In this particular case, as the code currently stands, it's not a problem...because the function never mutates `algorithms` (although, we cannot promise that the call to `token_introspect_offline()` doesn't mutate it...).  However, the code could change in the future, and it would be good to protect against problems from that.